index.json

```diff
@@ -550,6 +551,8 @@
     "react/self-closing-comp": "error",
     "react/sort-comp": "error",
     "react/sort-prop-types": "error",
+    "react-hooks/rules-of-hooks": "error",
+    "react-hooks/exhaustive-deps": "warn",
```

Comment:
@willopez Let's make this an `error` here, and then if we need to we can override all the new rules with `warn` inside of `reaction`, that way they aren't blockers right off the bat while we are getting used to them / seeing exactly how they function.

https://github.com/reactioncommerce/reaction/blob/master/package.json#L396

Reply:
Sounds good, updated.